domain/consensus/processes/pruningmanager/pruningmanager.go

```diff
@@ -1133,3 +1127,37 @@ func (pm *pruningManager) isPruningPointInPruningDepth(stagingArea *model.Stagin
 
 	return blockGHOSTDAGData.BlueScore() >= pruningPointHeader.BlueScore()+pm.pruningDepth, nil
 }
+
+func (pm *pruningManager) TrustedBlockAssociatedGHOSTDAGDataBlockHashes(stagingArea *model.StagingArea, blockHash *externalapi.DomainHash) ([]*externalapi.DomainHash, error) {
+	blockHashes := make([]*externalapi.DomainHash, 0, pm.k)
+	current := blockHash
+	isTrustedData := false
+	for i := externalapi.KType(0); i < pm.k+1; i++ {
```

Comment:
This might be splitting hairs, but pm.k+1 might overflow. Please write i <= pm.k instead